include/sys/sdt.h

```diff
@@ -55,15 +51,9 @@
  * twice, so it should not have side effects (e.g. something like:
  * "return (SET_ERROR(log_error(EINVAL, info)));" would log the error twice).
  */
-#define	SET_ERROR(err) \
-	(trace_zfs_set__error(__FILE__, __func__, __LINE__, err), err)
-
-#else
-
 #undef SET_ERROR
-#define	SET_ERROR(err) (err)
-
-#endif /* HAVE_DECLARE_EVENT_CLASS */
+#define	SET_ERROR(err) \
+	(__dprintf(__FILE__, __func__, __LINE__, "error %lu", err), err)
```

Comment:
This looks like it introduces a build failure when compiling the updated tracepoints.

```
In file included from ./arch/x86/include/asm/spinlock.h:4:0,
                 from include/linux/spinlock.h:87,
                 from include/linux/seqlock.h:35,
                 from include/linux/time.h:5,
                 from include/linux/stat.h:18,
                 from include/linux/module.h:10,
                 from /home/behlendo/src/git/spl/include/sys/sysmacros.h:28,
                 from /home/behlendo/src/git/spl/include/sys/types.h:29,
                 from /home/behlendo/src/git/zfs/include/sys/zfs_context.h:34,
                 from /home/behlendo/src/git/zfs/include/sys/multilist.h:22,
                 from /home/behlendo/src/git/zfs/module/zfs/trace.c:26:
include/linux/jump_label.h:120:40: error: initializer element is not constant
 #define STATIC_KEY_INIT_FALSE ((struct static_key) \
                                        ^
include/linux/tracepoint.h:210:24: note: in expansion of macro ‘STATIC_KEY_INIT_FALSE’
   { __tpstrtab_##name, STATIC_KEY_INIT_FALSE, reg, unreg, NULL };\
                        ^
include/linux/tracepoint.h:216:2: note: in expansion of macro ‘DEFINE_TRACE_FN’
  DEFINE_TRACE_FN(name, NULL, NULL);
  ^
include/trace/define_trace.h:45:2: note: in expansion of macro ‘DEFINE_TRACE’
  DEFINE_TRACE(name)
  ^
/home/behlendo/src/git/zfs/include/sys/trace_dbgmsg.h:59:1: note: in expansion of macro ‘DEFINE_EVENT’
 DEFINE_EVENT(zfs_dprintf_class, name, \
 ^
/home/behlendo/src/git/zfs/include/sys/trace_dbgmsg.h:64:1: note: in expansion of macro ‘DEFINE_DPRINTF_EVENT’
 DEFINE_DPRINTF_EVENT(zfs_zfs__dprintf);
```